app/models/service_template_ansible_playbook.rb

```diff
@@ -134,26 +144,55 @@ def validate_update_config_info(options)
     self.class.send(:validate_config_info, opts)
   end
 
-  def job_template(action)
+  def job_template!(action)
     resource_actions.find_by!(:action => action.to_s.capitalize).configuration_template
   end
 
-  def update_catalog_item(options, auth_user = nil)
-    config_info = validate_update_config_info(options)
-    name = options[:name]
-    description = options[:description]
+  def job_template(action)
+    resource = resource_actions.find_by(:action => action.to_s.capitalize)
+    resource.present? ? resource.configuration_template : nil
+  end
+
+  def create_dialogs(config_info)
     [:provision, :retirement, :reconfigure].each do |action|
-      next unless config_info[action]
       info = config_info[action]
+      if new_dialog_required?(info)
+        info[:dialog_id] = create_new_dialog(info[:new_dialog_name], job_template!(action), info[:hosts]).id
+      end
+    end
+  end
 
-      new_dialog = create_new_dialog(info[:new_dialog_name], job_template(action), info[:hosts]) if info[:new_dialog_name]
-      config_info[action][:dialog_id] = new_dialog.id if new_dialog
+  def new_job_templates(config_info, name, description, auth_user)
+    config_info.deep_merge!(send(:create_job_template, name, description, config_info, auth_user))
+  end
 
-      next unless info.key?(:playbook_id)
-      tower, params = self.class.send(:build_parameter_list, "miq_#{name}_#{action}", description, info)
-      params[:manager_ref] = job_template(action).manager_ref
-      ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ConfigurationScript.update_in_provider_queue(tower.id, params, auth_user)
+  def update_job_templates(name, description, config_info, auth_user)
+    [:provision, :retirement, :reconfigure].each do |action|
+      info = config_info[action]
+      next unless info
```

Comment:
Since `job_template` may not exist, we should never call `job_template!`. We do not need to update a job template if it does not exist. Therefore the logic should be like this
```
  next unless info   # no intention to change anything for this action
  
  job_template = job_template(action)
  next unless job_template   # job_template does not exist
  
  if info.key?(:playbook_id) # job_template exists, and playbook is given, try an update
    # update
  else                       # job_template exists but no more playbook, try a delete
    # delete
  end
```